me.id.webverify/webverifylib/src/main/java/me/id/webverifylib/IDmeWebVerify.java

```diff
@@ -243,9 +254,19 @@ private void SendDataBack(String response)
     {
         activity.getIntent().putExtra(IDME_WEB_VERIFY_RESPONSE, response);
         activity.setResult(Activity.RESULT_OK, activity.getIntent());
-
         activity.finish();
     }
 
-
+    /**
+     * Sends the error back to the activity that called the WebView.
+     *
+     * @param error The service error form the Web Request.
+     */
+    private void SendErrorBack(String error)
+    {
+        Log.e("Web Request Error", error);
+        activity.getIntent().putExtra(IDME_WEB_VERIFY_RESPONSE, error);
+        activity.setResult(Activity.RESULT_CANCELED);
```

Comment:
the problem is that in my opinion this way of communication is not the best, we have to change it to a logic with listeners. Before this change if an error occurred, the sdk send to the activity the `Activity.RESULT_OK` code, in my opinion canceled is better than ok, but I know that we should improve it.